src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1172 @@ const ActorInit En_Kakasi_InitVars = {
+                this->actor.textId = 0x165C; // "Would you like to learn it?  No Yes"
+                this->unkMsgState1AC = 4;
+
+            } else if (this->actor.textId == 0x165E) {
+                // "How's that? Pretty interesting, isn't it?  If you learn a new song ...
+                this->actor.textId = 0x165F;
+            } else {
+                EnKakasi_SetupIdleStanding(this);
+                return;
+            }
+        } else {
+            this->unkMsgState1AC = 5;
+
+            if (globalCtx->msgCtx.choiceIndex == 1) {
+                func_8019F208(); // play 0x4808 sfx (decide) and calls func_801A75E8
+                // "Do you want to learn it?  No Yes"
+                if (this->actor.textId == 0x1656) {
+                    // "Oh, yeah!  Now listen up!  If you play that strange song ...
+                    this->actor.textId = 0x1658;
+                // "Would you like to learn it?  No Yes"
+                } else if (this->actor.textId == 0x165C) {
+                    this->actor.textId = 0x165E;
+                } else if (saveContextDay == 3 && gSaveContext.isNight) {
+                    // "But outside it seems to have gotten kind of...dangerous.
+                    this->actor.textId = 0x164F;
+                } else {
+                    // "Oh, yeah!  In that case, forget the time.  Let's dance!"
+                    this->actor.textId = 0x1652;
+                }
+                EnKakasi_SetAnimation(this, ENKAKASI_ANIME_HOPPING_REGULAR);
+            } else {
+                func_8019F230(); // play 0x480A sfx (cancel) and calls func_801A75E8
+                if (this->actor.textId == 0x1656) { // "Do you want to learn it?  No Yes"
+                    // "Oh, sorry. That's too bad.  In that case, let me hear a song written by you, baby!"
+                    this->actor.textId = 0x1657;
+                } else if (this->actor.textId == 0x165C) { // "Would you like to learn it?  No Yes"
+                    // v "Oh, sorry! That's too bad.  In that case, see you later!"
+                    this->actor.textId = 0x165D;
+                } else {
+                    // "Oh, sorry. That's too bad. In that case, come back anytime if you do ...
+                    this->actor.textId = 0x1651;
+                }
+                this->unkCounter1A4 = 0;
+                if (this->animeIndex != ENKAKASI_ANIME_ARMS_CROSSED_ROCKING) {
+                    EnKakasi_SetAnimation(this, ENKAKASI_ANIME_ARMS_CROSSED_ROCKING);
+                }
+            }
+        }
+        func_80151938(globalCtx, this->actor.textId);
+    }
+}
+
+void EnKakasi_SetupSongTeach(EnKakasi* this, GlobalContext* globalCtx) {
+    // "Oh, baby!  That's a nice thing you got there!  Let me hear a song you wrote on that!"
+    this->actor.textId = 0x1646;
+    func_801518B0(globalCtx, this->actor.textId, &this->actor);
+    this->cutsceneCamId = 0; // SUBCAM_FREE
```

Comment:
```suggestion
    this->cutsceneCamId = 0;
```
As a side-note, I'll be taking care of all these macros in my Camera PR